api/odigos/v1alpha1/instrumentationrules/http_headers.go

```diff
@@ -0,0 +1,9 @@
+package instrumentationrules
+
+// +kubebuilder:object:generate=true
+// +kubebuilder:deepcopy-gen=true
+type HttpHeadersCollection struct {
+
+	// Limit payload collection to specific header keys.
+	HeaderKeys *[]string `json:"headerKeys,omitempty"`
```

Comment:
need to also add it in docs:

https://docs.odigos.io/pipeline/rules/introduction